# Análisis EDA y Decisiones de Negocio en Ventas de Automóviles

Este notebook contiene 15 ejercicios que combinan análisis exploratorio de datos (EDA) y reflexiones críticas de negocio. Cada punto incluye instrucciones detalladas, paso a paso, para realizar agrupaciones y filtros, de modo que se guíe al estudiante en cada parte del proceso.

La estructura del dataset es la siguiente:

```
year, make, model, trim, body, transmission, vin, state, condition, odometer, color, interior, seller, mmr, sellingprice, saledate
```

¡Comienza el análisis y sigue cada paso para proponer estrategias de negocio basadas en los resultados!

In [40]:
# Descargar y descomprimir el dataset
!wget https://github.com/javierherrera1996/lecture_analytics/raw/main/cars_dataset.zip
!unzip -o cars_dataset.zip

--2025-03-13 21:23:37--  https://github.com/javierherrera1996/lecture_analytics/raw/main/cars_dataset.zip
Resolving github.com (github.com)... 140.82.112.3
Connecting to github.com (github.com)|140.82.112.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/javierherrera1996/lecture_analytics/main/cars_dataset.zip [following]
--2025-03-13 21:23:37--  https://raw.githubusercontent.com/javierherrera1996/lecture_analytics/main/cars_dataset.zip
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 19753181 (19M) [application/zip]
Saving to: ‘cars_dataset.zip.1’

cars_dataset.zip.1  100%[===================>]  18.84M  --.-KB/s    in 0.09s   

2025-03-13 21:23:37 (202 MB/s) - ‘cars_dataset.zip.1’ saved [19753

## 1. Dimensiones y Representatividad del Dataset

a) Carga el dataset (desde un archivo CSV) y muestra sus dimensiones (número de filas y columnas).

b) Describe brevemente qué representa cada fila y reflexiona: ¿Crees que el dataset es representativo del mercado de ventas de automóviles? Justifica tu respuesta.

In [41]:
import pandas as pd

data = pd.read_csv('car_prices.csv')


In [42]:
data.shape

(558837, 16)

In [43]:
data.head()

year   make                model        trim   body transmission  \
0  2015    Kia              Sorento          LX    SUV    automatic   
1  2015    Kia              Sorento          LX    SUV    automatic   
2  2014    BMW             3 Series  328i SULEV  Sedan    automatic   
3  2015  Volvo                  S60          T5  Sedan    automatic   
4  2014    BMW  6 Series Gran Coupe        650i  Sedan    automatic   

                 vin state  condition  odometer  color interior  \
0  5xyktca69fg566472    ca        5.0   16639.0  white    black   
1  5xyktca69fg561319    ca        5.0    9393.0  white    beige   
2  wba3c1c51ek116351    ca       45.0    1331.0   gray    black   
3  yv1612tb4f1310987    ca       41.0   14282.0  white    black   
4  wba6b2c57ed129731    ca       43.0    2641.0   gray    black   

                                   seller      mmr  sellingprice  \
0                 kia motors america  inc  20500.0       21500.0   
1                 kia motors america  inc  20800.0       21500.0   
2  financial services remarketing (lease)  31900.0       30000.0   
3                 volvo na rep/world omni  27500.0       27750.0   
4  financial services remarketing (lease)  66000.0       67000.0   

                                  saledate  
0  Tue Dec 16 2014 12:30:00 GMT-0800 (PST)  
1  Tue Dec 16 2014 12:30:00 GMT-0800 (PST)  
2  Thu Jan 15 2015 04:30:00 GMT-0800 (PST)  
3  Thu Jan 29 2015 04:30:00 GMT-0800 (PST)  
4  Thu Dec 18 2014 12:30:00 GMT-0800 (PST)

In [44]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 558837 entries, 0 to 558836
Data columns (total 16 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   year          558837 non-null  int64  
 1   make          548536 non-null  object 
 2   model         548438 non-null  object 
 3   trim          548186 non-null  object 
 4   body          545642 non-null  object 
 5   transmission  493485 non-null  object 
 6   vin           558833 non-null  object 
 7   state         558837 non-null  object 
 8   condition     547017 non-null  float64
 9   odometer      558743 non-null  float64
 10  color         558088 non-null  object 
 11  interior      558088 non-null  object 
 12  seller        558837 non-null  object 
 13  mmr           558799 non-null  float64
 14  sellingprice  558825 non-null  float64
 15  saledate      558825 non-null  object 
dtypes: float64(4), int64(1), object(11)
memory usage: 68.2+ MB


Si, es representativo ya que tienen los precios del venta, modelos, años y demas info que es importante para el mercado en cuanto a la venta de automoviles

## 2. Ventas Totales y Promedio

a) Calcula la suma total y el promedio de la columna `sellingprice`.

b) Caso de negocio: Si el precio promedio es bajo, ¿qué estrategias de precio o promociones sugerirías para aumentar la rentabilidad? Explica tu respuesta.

In [45]:
total_ventas= data['sellingprice'].sum()
prom_ventas = data['sellingprice'].mean()

print(f"La suma total de ventas es: {total_ventas}")
print(f"El promedio de ventas es: {prom_ventas}")

La suma total de ventas es: 7606367587.0
El promedio de ventas es: 13611.358810003132


De acuerdo a este bajo precio podriamos implementar estrategias en cuanto a precios, ofrecer precios un poco mas accesibles para los clientes con esto llegar a más público, tambien jugar con promociones por tiempo limitado y estacionales con diferentes tipos de carros para impulsar su venta y como adicional implementar metodos de financiacion que ayude a que el cliente tenga facilidades de pago.

## 3. Distribución de Ventas por Marca

a) Utiliza `value_counts()` para determinar cuántas ventas se registraron por cada marca (`make`).

b) Caso de negocio: ¿Qué marcas parecen dominar el mercado? Basado en esta distribución, ¿en qué marcas invertirías en publicidad y por qué?

In [46]:
data['make'].value_counts()

make
Ford          93554
Chevrolet     60197
Nissan        53946
Toyota        39871
Dodge         30710
              ...  
airstream         1
ford tk           1
chev truck        1
hyundai tk        1
Lotus             1
Name: count, Length: 96, dtype: int64

Las marcas dominantes son ford(93554) chevrolet (60197) y nissan (53946), en las marcas en las que invirtia son ford, chevrolet y nissan que ya tiene presencia en el mercado y buenos numeros y en cuanto a publicidad ya que es una grna inversion a las marcas que no tienen tanta presencia en el mercado para potenciar sus ventas.

## 4. Segmentación de Clientes por Kilometraje

a) Crea un DataFrame filtrado que incluya únicamente las ventas donde el `odometer` es menor a 20,000 millas. Cuantos Vehiculos tiene este dataset?

b) Caso de negocio: Este segmento de vehículos con bajo uso puede tener mayor valor percibido. ¿Cómo ajustarías la estrategia de marketing para este grupo?

In [47]:
data_filtered = data[data['odometer'] < 20000]
print(data_filtered.head())
print(data_filtered.columns)

   year   make                model        trim   body transmission  \
0  2015    Kia              Sorento          LX    SUV    automatic   
1  2015    Kia              Sorento          LX    SUV    automatic   
2  2014    BMW             3 Series  328i SULEV  Sedan    automatic   
3  2015  Volvo                  S60          T5  Sedan    automatic   
4  2014    BMW  6 Series Gran Coupe        650i  Sedan    automatic   

                 vin state  condition  odometer  color interior  \
0  5xyktca69fg566472    ca        5.0   16639.0  white    black   
1  5xyktca69fg561319    ca        5.0    9393.0  white    beige   
2  wba3c1c51ek116351    ca       45.0    1331.0   gray    black   
3  yv1612tb4f1310987    ca       41.0   14282.0  white    black   
4  wba6b2c57ed129731    ca       43.0    2641.0   gray    black   

                                   seller      mmr  sellingprice  \
0                 kia motors america  inc  20500.0       21500.0   
1                 kia motors ameri

Lo primero es ajustar la estrategia de marketing ya que son autos viejos y tiene su uso entonce podria implementar algun tipo de descuento atractivo, igual intentar vender los autos con algunas experiencias de clientes que han tenido autos similares y la capacidad de los autos de recorrer largas distancias haran que las ersonas se interecen por este.



## 5. Agrupación por Marca: Precio y Volumen

a) Agrupa el dataset por `make` y calcula:
   - El promedio de `sellingprice`.
   - La cantidad de ventas por marca (usa el conteo de registros como proxy).

b) Caso de negocio: Si una marca tiene alto volumen pero precios bajos, ¿qué medidas recomendarías para mejorar el margen?

**Paso a paso:**
1. Realiza la agrupación.
2. Renombra las columnas para que sean claras.
3. Interpreta el resultado antes de pasar al siguiente filtro o análisis.

In [48]:
prom_sellingprice= data.groupby('make').agg({'sellingprice': 'mean', 'make': 'count'})
prom_sellingprice= prom_sellingprice.rename(columns={'sellingprice': 'average_sellingprice', 'make': 'sales_count'})
prom_sellingprice

average_sellingprice  sales_count
make                                           
Acura                 14017.268260         5901
Aston Martin          54812.000000           25
Audi                  19915.432782         5869
BMW                   21441.895748        20719
Bentley               74367.672414          116
...                            ...          ...
subaru                 3710.416667           60
suzuki                 4810.000000            5
toyota                 7339.105263           95
volkswagen             6145.833333           24
vw                    13672.916667           24

[96 rows x 2 columns]

La esrategia que implementaria es interara exaltar sus beneficio spara poderlas vender como marcas premium o exclusivas y que asi se pueda justificar su valor , en cuanto a markering un rebranding de la marca para poderla posicionar en un mercado premium, por medio de las marcas dar a conocer esos carros con experiencias y por medio de tendencias generar influencia por redes sociales

## 6. Análisis de Rentabilidad por Marca

a) Crea una nueva columna `profit` que sea la diferencia entre `mmr` y `sellingprice`.
b) Agrupa por `make` y calcula el beneficio promedio para cada marca.

c) Caso de negocio: Identifica las 5 marcas con mayor beneficio promedio y explica qué acciones estratégicas implementarías (por ejemplo, renegociar precios o invertir en calidad).

In [49]:
data['profit'] = data['mmr'] - data['sellingprice']

benefi_prom= data.groupby('make')['profit'].mean()
print(benefi_prom)

top_5_brands = benefi_prom.sort_values(ascending=False).head(5)
print("\nTop 5 brands with the highest average profit:")
top_5_brands


make
Acura             59.544992
Aston Martin   -1252.000000
Audi             164.781053
BMW              133.652058
Bentley         1560.775862
                   ...     
subaru           201.250000
suzuki            -5.000000
toyota            28.526316
volkswagen      -161.458333
vw               175.000000
Name: profit, Length: 96, dtype: float64

Top 5 brands with the highest average profit:


make
Fisker      2638.888889
Ferrari     2000.000000
Plymouth    1577.500000
Bentley     1560.775862
porsche     1419.736842
Name: profit, dtype: float64

Las marcas con mayor beneficio son fidker, ferrari,plymouth, bentley y porshe estas tienen oportunidades de serguir inviertiendo in innovaciom, mejorando la xperienia del cliente, tener epeeriencias personalilzadas, ediciones limitadas, apuntas a la sostenibilidad y explorar nuevos nichos de mercados

## 7. Análisis Temporal: Ventas del Año 2015

a) Filtra el dataset para obtener únicamente las ventas del año 2015 y muestra las primeras 10 filas. Cuantos Vehiculos tiene esta dataset


b) Caso de negocio: Si 2015 muestra comportamientos atípicos en comparación con otros años, ¿deberías ajustar la estrategia basándote en tendencias actuales? Explica tu criterio.

In [50]:
ventas_2015 = data[data['year'] == 2015]
print(ventas_2015.head(10))
num_vehiculos_2015 = len(ventas_2015)
print(f"\nNúmero de vehículos vendidos en 2015: {num_vehiculos_2015}")

    year       make    model   trim   body transmission                vin  \
0   2015        Kia  Sorento     LX    SUV    automatic  5xyktca69fg566472   
1   2015        Kia  Sorento     LX    SUV    automatic  5xyktca69fg561319   
3   2015      Volvo      S60     T5  Sedan    automatic  yv1612tb4f1310987   
5   2015     Nissan   Altima  2.5 S  Sedan    automatic  1n4al3ap1fn326013   
11  2015        Kia   Optima     LX  Sedan    automatic  5xxgm4a73fg353538   
12  2015       Ford   Fusion     SE  Sedan    automatic  3fa6p0hdxfr145753   
13  2015        Kia  Sorento     LX    SUV    automatic  5xyktca66fg561407   
15  2015     Nissan   Altima  2.5 S  Sedan    automatic  1n4al3ap5fc124223   
16  2015    Hyundai   Sonata     SE  Sedan    automatic  5npe24af4fh001562   
20  2015  Chevrolet   Impala    LTZ  Sedan    automatic  2g1165s30f9103921   

   state  condition  odometer   color interior  \
0     ca        5.0   16639.0   white    black   
1     ca        5.0    9393.0   white    

Si mostrase comportamientos atipicos buscaria que paso en ese año en el contexto que se desarrollaron esos datos atipicios para entendere que paso y poder buscar alguna situacion similar que me ayude a entender y tomar decisiones coherentes.

## 8. Comparación por Tipo de Carrocería

a) Agrupa por la columna `body` y calcula el promedio de `sellingprice` para cada tipo.

b) Caso de negocio: Determina qué tipo de carrocería tiene mayor precio promedio y argumenta cómo podrías aprovechar esta información para posicionar una línea premium o ajustar el inventario.

In [51]:
data_grouped = data.groupby('body')['sellingprice'].mean().reset_index()
data_grouped.rename(columns={'sellingprice': 'avg_sellingprice'}, inplace=True)
print(data_grouped)

                  body  avg_sellingprice
0           Access Cab      13262.500000
1   Beetle Convertible      18070.192308
2            CTS Coupe      22758.527132
3            CTS Wagon      17811.538462
4          CTS-V Coupe      38425.750000
..                 ...               ...
82         transit van      22707.142857
83     tsx sport wagon      18118.750000
84                 van      15009.742105
85               wagon      10551.605442
86             xtracab       5250.000000

[87 rows x 2 columns]


Se podria implementar una estrategia con la marca de CTS en la que se desarrolle un alinea premium con caracteristicas similares, incrementar el stock disponible en este y reducir el de carros de bajo valor con caracteristicas similares.

## 9. Filtros Combinados para un Caso Específico

a) **Paso 1:** Agrupa los datos por `state`, `model` y calcula el promedio de `sellingprice` para ver cuál es el precio promedio por combinación.

b) **Paso 2:** Utiliza el resultado del agrupamiento para filtrar el dataset original. Por ejemplo, filtra los registros de un estado ('ca') y un modelo ('Sorento') cuyo `sellingprice` supere el precio promedio calculado.

c) Caso de negocio: Analiza este segmento y plantea qué acciones implementarías para potenciar las ventas en California, considerando el modelo y el segmento premium.

In [52]:
data_grouped = data.groupby(['state', 'model'])['sellingprice'].mean().reset_index()
data_grouped.rename(columns={'sellingprice': 'avg_sellingprice'}, inplace=True)

ca_sorento_avg = data_grouped[(data_grouped['state'] == 'ca') & (data_grouped['model'] == 'Sorento')]['avg_sellingprice'].values[0]

data_filtered = data[(data['state'] == 'ca') & (data['model'] == 'Sorento') & (data['sellingprice'] > ca_sorento_avg)]

print(data_filtered.head(10))

    year make    model trim body transmission                vin state  \
0   2015  Kia  Sorento   LX  SUV    automatic  5xyktca69fg566472    ca   
1   2015  Kia  Sorento   LX  SUV    automatic  5xyktca69fg561319    ca   
13  2015  Kia  Sorento   LX  SUV    automatic  5xyktca66fg561407    ca   
32  2015  Kia  Sorento   LX  SUV    automatic  5xyktca60fg565226    ca   
34  2015  Kia  Sorento   LX  SUV    automatic  5xyktca68fg559481    ca   
49  2015  Kia  Sorento   LX  SUV    automatic  5xyktca67fg570973    ca   
55  2015  Kia  Sorento   LX  SUV    automatic  5xyktca63fg570727    ca   
58  2015  Kia  Sorento   LX  SUV    automatic  5xykt3a62fg579841    ca   
65  2015  Kia  Sorento   LX  SUV    automatic  5xyktca6xfg560888    ca   
78  2015  Kia  Sorento   LX  SUV    automatic  5xyktca68fg570979    ca   

    condition  odometer   color interior                   seller      mmr  \
0         5.0   16639.0   white    black  kia motors america  inc  20500.0   
1         5.0    9393.0   whi

Para potenciar las vntas KIA en California, se debe  enfocar en el segmento premium, ajustando precios de venta, ofrercer paquetes adicionales como algun mantenimiiento en cierto tiempo, hacer campañas de marketing sirigidas a esta marca de lujo donde se vea su diseño, espacio interior y se cree la necesidad en el publico obj.

## 10. Ventas Totales por Estado

a) Agrupa por `state` y calcula la suma de `sellingprice` para cada estado.

b) Caso de negocio: Identifica el estado con mayores ventas totales y discute cómo enfocar campañas de marketing o ampliar la red de distribución en ese estado.

In [53]:
data = data[data['state'].str.len() == 2]
data_grouped = data.groupby('state', as_index=False)['sellingprice'].sum()
data_grouped.rename(columns={'sellingprice': 'total_sellingprice'}, inplace=True)
print(data_grouped)

   state  total_sellingprice
0     ab        1.036765e+07
1     al        2.377000e+05
2     az        1.009105e+08
3     ca        1.061078e+09
4     co        1.234499e+08
5     fl        1.151065e+09
6     ga        4.489274e+08
7     hi        1.508005e+07
8     il        3.472900e+08
9     in        4.634485e+07
10    la        2.305286e+07
11    ma        4.526425e+07
12    md        8.267062e+07
13    mi        2.309001e+08
14    mn        1.292545e+08
15    mo        2.316471e+08
16    ms        2.081288e+07
17    nc        1.891986e+08
18    ne        5.045528e+07
19    nj        3.781154e+08
20    nm        1.075900e+06
21    ns        6.230000e+05
22    nv        1.915152e+08
23    ny        6.482348e+07
24    oh        3.108228e+08
25    ok        4.981000e+05
26    on        6.131105e+07
27    or        1.335005e+07
28    pa        8.612346e+08
29    pr        2.564435e+07
30    qc        1.654975e+07
31    sc        4.389656e+07
32    tn        3.554186e+08
33    tx      

El estado con mayor ventas es California con 1.061.078.000 en cuanto a ampliar redes de ditribucion en california primero dee hacerse estrategias de marketing en redes sociales para llegar a un gran publico en el estado, tener algunos eventos donde se hagan pruebas de maneo y demas actividades que hagen ver la marca, y donde de igual manera se inaguren clocales, en lugares estartaegicos como centros comerciales.

## 11. Rentabilidad por Estado

a) Agrupa por `state` y calcula el promedio de eficiencia operativa. Como no contamos con 'Operating Margin', usa como proxy `profit/sellingprice`.

b) Caso de negocio: Identifica el estado con mayor eficiencia operativa y sugiere estrategias para replicar ese éxito en otros mercados.

In [54]:
data['prom_eficiencia'] = data['profit'] / data['sellingprice']
estado_eficiencia = data.groupby('state')['prom_eficiencia'].mean()

estado_eficiencia

<ipython-input-54-32075b381f9f>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['prom_eficiencia'] = data['profit'] / data['sellingprice']


state
ab    0.070770
al    0.169054
az    0.108662
ca    0.164543
co    0.022271
fl    0.069392
ga    0.247233
hi    0.429630
il    0.063233
in    0.209087
la    0.219902
ma    0.291411
md    0.292650
mi    0.061924
mn    0.031394
mo    0.040832
ms    0.223003
nc    0.032058
ne    0.068946
nj    0.115883
nm    1.178797
ns    0.104358
nv    0.082634
ny    0.320112
oh    0.136039
ok    0.208784
on    0.018810
or    0.104368
pa    0.047799
pr    0.276526
qc    0.071734
sc    0.078001
tn    0.039025
tx    0.565770
ut    0.284350
va    0.202064
wa    0.042004
wi    0.150673
Name: prom_eficiencia, dtype: float64

Quien tiene la mayor eficiencia es Nuevo Mexico, para aplicar estas estrategias primero se debe hacer un analisis de las practicas que tienen alli para poderlas replicar en otros estados, luego de esta capacitar al personal en estas nuevas practicas

## 12. Análisis por Tipo de Transmisión (en lugar de Sales Method)

a) Agrupa el dataset por `transmission` y calcula:
   - La suma de `sellingprice` para cada tipo.
   - El promedio de `sellingprice` para cada tipo.

b) Caso de negocio: ¿Qué tipo de transmisión (automática vs. manual) genera mayores ingresos y qué implicaciones tiene para la estrategia comercial?

In [55]:
trans_analisis= data.groupby('transmission')['sellingprice'].agg(['sum', 'mean'])
trans_analisis

sum          mean
transmission                            
automatic     6.443934e+09  13540.406866
manual        1.966936e+08  11211.448358

La transmision automatica genera mayores ingresos, prioriza los inventarios de la automatica dejando de lado los manuales se debe hacer promoción y aumentando el stock de la automatica.

## 13. Segmentación por Producto Líder

a) Identifica el producto (utiliza la columna `model` como proxy) con mayor cantidad de unidades vendidas. Agrupa por `model` y usa el conteo de registros.

b) Caso de negocio: Si un producto lidera en ventas, ¿cómo ajustarías el inventario y las promociones para maximizar las ganancias?

In [74]:
model_ventas = data.groupby('model')['model'].count().reset_index(name='count')
leading_product = model_ventas.sort_values('count', ascending=False).iloc[0]
leading_product

model    Altima
count     19349
Name: 76, dtype: object

Se debe ajustar el inventario teniendo en cuenta la demanda, implementar descuentos y promociones , hacer un tipo de experiencia premiun al cliente y metodos de financiacion atractivos.

## 14. Comparación Crítica entre Retailers

a) Compara dos retailers usando la columna `seller` en cuanto a volumen de ventas y margen de beneficio (usa `profit` como proxy).

b) Caso de negocio: Si un retailer muestra mayores unidades vendidas pero un margen inferior, ¿qué estrategia recomendarías (por ejemplo, optimización de costos, cambio en política de descuentos, etc.)? Justifica tu propuesta.

**Paso a paso:**
1. Agrupa por `seller` y calcula la suma de `sellingprice` y el promedio de `profit`.
2. Ordena los resultados para identificar los líderes en volumen y analiza el margen.
3. Propón una estrategia basada en los resultados.

In [62]:
retailer_comp = data.groupby('seller').agg({'sellingprice': 'sum', 'profit': 'mean'})
retailer_comp = retailer_comp.sort_values(by='sellingprice', ascending=False)

retailer_comp

sellingprice      profit
seller                                                    
ford motor credit company llc      340960125.0  216.393122
nissan-infiniti lt                 272104560.0  -70.225969
the hertz corporation              250863168.0  189.344063
nissan infiniti lt                 216158321.0   92.654989
avis corporation                   203432330.0  -20.590510
...                                        ...         ...
alberta truck & auto liquidators         200.0  150.000000
district 24c lions charity               200.0  325.000000
republic finance llc/pearl               200.0  150.000000
preston ford inc                         200.0  750.000000
kfl llc                                    0.0         NaN

[14261 rows x 2 columns]

Se necesita implementar estrategias de rentabilidad en las que se pueda mejorar, donde prevalezca disminuir costos, en este momento disminuir descuentos y algunos gastos que hagan bajar la rentabilidad total

## 15. Caso Integrador: Estrategia Comercial Global

a) **Paso 1:** Identifica cada criterio basado en los análisis anteriores:
   - **Retailer con mayores ventas totales:** Usa la columna `seller` del resumen de retailers.
   - **Estado con mayores ventas totales:** Del agrupamiento por `state`.
   - **Producto con mayor unidades vendidas:** Usa la columna `model`.
   - **Tipo de transmisión con mayores ventas totales:** Agrupa por `transmission`.

b) **Paso 2:** Filtra el dataset original usando cada uno de estos criterios, uno a uno, y luego combínalos en un filtro final.

c) Con base en el análisis combinado, redacta un breve informe (3-4 líneas) proponiendo una estrategia comercial integral para maximizar la rentabilidad, considerando inversión en canales, promociones específicas y optimización de inventario.

d) Caso de negocio: Utiliza la integración de estos criterios para definir en qué segmento enfocar la estrategia global.

In [72]:

top_seller = data.groupby('seller')['sellingprice'].sum().idxmax()

top_state = data.groupby('state')['sellingprice'].sum().idxmax()

top_model = data['model'].value_counts().idxmax()

top_transmission = data.groupby('transmission')['sellingprice'].sum().idxmax()


data_filtrado = data[
    (data['seller'] == top_seller) &
    (data['state'] == top_state) &
    (data['model'] == top_model) &
    (data['transmission'] == top_transmission)
]

data_filtered.head(10)

year make    model trim body transmission                vin state  \
0   2015  Kia  Sorento   LX  SUV    automatic  5xyktca69fg566472    ca   
1   2015  Kia  Sorento   LX  SUV    automatic  5xyktca69fg561319    ca   
13  2015  Kia  Sorento   LX  SUV    automatic  5xyktca66fg561407    ca   
32  2015  Kia  Sorento   LX  SUV    automatic  5xyktca60fg565226    ca   
34  2015  Kia  Sorento   LX  SUV    automatic  5xyktca68fg559481    ca   
49  2015  Kia  Sorento   LX  SUV    automatic  5xyktca67fg570973    ca   
55  2015  Kia  Sorento   LX  SUV    automatic  5xyktca63fg570727    ca   
58  2015  Kia  Sorento   LX  SUV    automatic  5xykt3a62fg579841    ca   
65  2015  Kia  Sorento   LX  SUV    automatic  5xyktca6xfg560888    ca   
78  2015  Kia  Sorento   LX  SUV    automatic  5xyktca68fg570979    ca   

    condition  odometer   color interior                   seller      mmr  \
0         5.0   16639.0   white    black  kia motors america  inc  20500.0   
1         5.0    9393.0   white    beige  kia motors america  inc  20800.0   
13        5.0   14634.0  silver    black  kia motors america  inc  20600.0   
32        5.0   13757.0     red    black  kia motors america  inc  20600.0   
34       44.0   12862.0    gray    black  kia motors america  inc  20700.0   
49        5.0   13878.0  silver    black  kia motors america  inc  20600.0   
55       48.0   10357.0  silver     gray  kia motors america  inc  20700.0   
58        5.0    5214.0   white    black       kia motors finance  19400.0   
65        5.0   14176.0   white    black  kia motors america  inc  20600.0   
78       44.0    9161.0  silver    black  kia motors america  inc  20800.0   

    sellingprice                                 saledate  profit  
0        21500.0  Tue Dec 16 2014 12:30:00 GMT-0800 (PST) -1000.0  
1        21500.0  Tue Dec 16 2014 12:30:00 GMT-0800 (PST)  -700.0  
13       21500.0  Tue Dec 16 2014 12:30:00 GMT-0800 (PST)  -900.0  
32       20750.0  Tue Dec 16 2014 12:30:00 GMT-0800 (PST)  -150.0  
34       21000.0  Tue Dec 16 2014 12:30:00 GMT-0800 (PST)  -300.0  
49       20750.0  Tue Dec 16 2014 12:30:00 GMT-0800 (PST)  -150.0  
55       20500.0  Tue Dec 16 2014 12:30:00 GMT-0800 (PST)   200.0  
58       19300.0  Tue Dec 16 2014 12:00:00 GMT-0800 (PST)   100.0  
65       21250.0  Tue Dec 16 2014 12:30:00 GMT-0800 (PST)  -650.0  
78       22250.0  Tue Dec 16 2014 12:30:00 GMT-0800 (PST) -1450.0

Estrategia comercial se deberia hacer una inversion en cuanto a estrategias de markeing en canales fisicos y digitales dirigidas a la marca y a la apertura de nuevos concesionarios.Tambien promocione especificas solo para modelos premium y solo estos tienen financiamientos especiales y por ultimo aumentar stocks de modelos rentables como los SUVS ajustando la oferta a la demanda local.


La estrategia es enfocarla en SUVS Automaticos y en cuanto a marca en especial a los KIA Sorento